storage/tools/dump_tree/main.go

```diff
@@ -0,0 +1,201 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// The dump_tree program uses the in memory storage implementation to create a sequenced
+// log tree of a particular size using known leaf data and then dumps out the resulting
+// SubTree protos for examination and debugging. It does not require any actual storage
+// to be configured.
+package main
+
+import (
+	"context"
+	"crypto"
+	"crypto/sha256"
+	"flag"
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/golang/protobuf/ptypes/any"
+	"github.com/google/trillian"
+	tc "github.com/google/trillian/crypto"
+	"github.com/google/trillian/crypto/sigpb"
+	"github.com/google/trillian/log"
+	"github.com/google/trillian/merkle/rfc6962"
+	"github.com/google/trillian/monitoring"
+	"github.com/google/trillian/quota"
+	"github.com/google/trillian/storage"
+	"github.com/google/trillian/storage/memory"
+	"github.com/google/trillian/storage/storagepb"
+	"github.com/google/trillian/util"
+	"github.com/google/trillian/crypto/keyspb"
+	"github.com/google/trillian/crypto/keys"
+	"github.com/golang/protobuf/ptypes"
+	"crypto/x509"
+)
+
+var (
+	treeSizeFlag       = flag.Int("tree_size", 871, "The number of leaves to be added to the tree")
+	batchSizeFlag      = flag.Int("batch_size", 50, "The batch size for sequencing")
+	leafDataFormatFlag = flag.String("leaf_format", "Leaf %d", "The format string for leaf data")
+)
+
+// This is a copy of the logserver private key from the testdata directory
+var logPrivKeyPEM string = `
```

Comment:
I thought it might be best to make this completely standalone so it can be run from anywhere.